primefaces/src/main/java/org/primefaces/component/selectcheckboxmenu/SelectCheckboxMenuRenderer.java

```diff
@@ -111,8 +127,7 @@ protected void encodeInputs(FacesContext context, SelectCheckboxMenu menu, List<
         writer.writeAttribute("class", "ui-helper-hidden", null);
 
         int idx = -1;
-        for (int i = 0; i < selectItems.size(); i++) {
-            SelectItem selectItem = selectItems.get(i);
+        for (SelectItem selectItem : selectItems) {
```

Comment:
https://github.com/primefaces/primefaces/wiki/Contributing-to-PrimeFaces#performance-guidelines